File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.js
function solution(data, col, row_begin, row_end) {
    
    const index = col-1;
    
    data.sort((a,b) => {
        if(a[index] == b[index]){
            return b[0] - a[0];
        }else{
            return a[index] - b[index];
        }
    })
    
    var answer = 0;
    
    for(let i=row_begin-1; i<row_end; i++){
        let sum = 0;
        data[i].forEach((num) => {
            sum += num%(i+1)
        })
        answer = (answer^sum)
    }
    
    return answer;
}

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.*;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        
        int index = col-1;
        
        Arrays.sort(data, new Comparator<int[]>(){
            @Override
            public int compare(int o1[], int o2[]){
                if(o1[index] == o2[index]){
                    return o2[0]-o1[0];
                }else{
                    return o1[index] - o2[index];
                }
            }
        });
        
        int answer = 0;
        
        for(int i=row_begin-1; i<row_end; i++){
            int sum = 0;
            for(int num : data[i]){
                sum += num % (i+1);
            }
            answer = (answer^sum);
        }

        return answer;
    }
}